integration/query_test.go

```diff
@@ -30,238 +30,6 @@ import (
 	"github.com/FerretDB/FerretDB/integration/shareddata"
 )
 
-func TestQuerySort(t *testing.T) {
-	t.Skip("https://github.com/FerretDB/FerretDB/issues/457")
-
-	t.Parallel()
-	ctx, collection := setup.Setup(t, shareddata.Scalars, shareddata.Composites)
-
-	for name, tc := range map[string]struct {
-		sort        bson.D
-		expectedIDs []any
-	}{
-		"Asc": {
-			sort: bson.D{{"v", 1}, {"_id", 1}},
-			expectedIDs: []any{
-				"array-empty",
-				"array-null",
-				"array-three",
-				"array-three-reverse",
-				"null",
-
-				"int64-min",
-				"int32-min",
-				"double-zero",
-				"int32-zero",
-				"int64-zero",
-				"double-smallest",
-				"array",
-				"double-whole",
-				"int32",
-				"int64",
-				"double",
-				"int32-max",
-				"int64-max",
-				"double-max",
-				"string-empty",
-				"string-whole",
-				"string-double",
-				"string",
-				"document-empty",
-				"document-null",
-				"document",
-				"document-composite",
-				"document-composite-reverse",
-				"binary-empty",
-				"binary",
-				"objectid-empty",
-				"objectid",
-				"bool-false",
-				"bool-true",
-				"datetime-year-min",
-				"datetime-epoch",
-				"datetime",
-				"datetime-year-max",
-				"timestamp-i",
-				"timestamp",
-				"regex-empty",
-				"regex",
-			},
-		},
-		"Desc": {
-			sort: bson.D{{"v", -1}, {"_id", 1}},
-			expectedIDs: []any{
-				"regex",
-				"regex-empty",
-				"timestamp",
-				"timestamp-i",
-				"datetime-year-max",
-				"datetime",
-				"datetime-epoch",
-				"datetime-year-min",
-				"bool-true",
-				"bool-false",
-				"objectid",
-				"objectid-empty",
-				"binary",
-				"binary-empty",
-				"document-composite-reverse",
-				"document-composite",
-				"document",
-				"document-null",
-				"document-empty",
-				"array-three",
-				"array-three-reverse",
-				"string",
-				"string-double",
-				"string-whole",
-				"string-empty",
-				"double-max",
-				"int64-max",
-				"int32-max",
-				"double",
-				"array",
-				"double-whole",
-				"int32",
-				"int64",
-				"double-smallest",
-				"double-zero",
-				"int32-zero",
-				"int64-zero",
-				"int32-min",
-				"int64-min",
-
-				"array-null",
-				"null",
-				"array-empty",
-			},
-		},
-	} {
-		name, tc := name, tc
-		t.Run(name, func(t *testing.T) {
-			t.Parallel()
-
-			cursor, err := collection.Find(ctx, bson.D{}, options.Find().SetSort(tc.sort))
-			require.NoError(t, err)
-
-			var actual []bson.D
-			err = cursor.All(ctx, &actual)
-			require.NoError(t, err)
-			assert.Equal(t, tc.expectedIDs, CollectIDs(t, actual))
-		})
-	}
-}
-
-// TODO: https://github.com/FerretDB/FerretDB/issues/636
-func TestQuerySortValue(t *testing.T) {
-	setup.SkipForTigris(t)
-
-	ctx, collection := setup.Setup(t, shareddata.Scalars)
-
-	for name, tc := range map[string]struct {
-		sort        bson.D
-		expectedIDs []any
-		err         *mongo.CommandError
-	}{
-		"AscValueScalar": {
-			sort: bson.D{{"v", 1}, {"_id", 1}},
-			expectedIDs: []any{
-				"null",
```

Comment:
It's quite tricky because `null` and `unset` are the equal value according to sort order, so I see they come in different order sometimes. Need test to not care about if `null` or `unset` comes in wrong order.

https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#non-existent-fields